### PR TITLE
SearchRepositoryByName improvements and unification (#6897)

### LINKED ISSUE #7001: 500/RuntimeError while browsing organization repo list
- Gitea version (or commit ref): 1.8.1
- Git version: 2.21.0.windows.1
- Operating system: Windows Server 2019
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [X] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [X] No
  - [ ] Not relevant
- Log gist: https://gist.github.com/philippbeckmann/5b9da6b2f45e5c940095a3dd4fa4d0de

## Description

When browsing an organization and switching to another page, an error 500 is displayed. The log reports "runtime error: invalid memory address or nil pointer dereference" (see gist).

### PR DESCRIPTION
**Pull request by lafriks**
Backport #6897, should fix #7001

### CODE REVIEW

File: models/repo_list_test.go [hunk 1]
```diff
@@ -150,7 +150,7 @@ func TestSearchRepositoryByName(t *testing.T) {
 			count: 19},
 		{name: "AllPublic/PublicAndPrivateRepositoriesOfUserIncludingCollaborative",
 			opts:  &SearchRepoOptions{Page: 1, PageSize: 10, OwnerID: 15, Private: true, AllPublic: true},
-			count: 24},
+			count: 27},
```
Comment (zeripath) [inline models/repo_list_test.go]:
```suggestion
			count: 25},
```

Two public repos were added in `models/fixtures/repository.yml`

File: integrations/api_repo_test.go [hunk 2]
```diff
@@ -69,40 +69,41 @@ func TestAPISearchRepo(t *testing.T) {
 		name, requestURL string
 		expectedResults
 	}{
-		{name: "RepositoriesMax50", requestURL: "/api/v1/repos/search?limit=50", expectedResults: expectedResults{
-			nil:   {count: 19},
-			user:  {count: 19},
-			user2: {count: 19}},
+		{name: "RepositoriesMax50", requestURL: "/api/v1/repos/search?limit=50&private=false", expectedResults: expectedResults{
+			nil:   {count: 21},
```
Comment (zeripath) [inline integrations/api_repo_test.go]:
```suggestion
			nil:   {count: 19},
```

File: integrations/api_repo_test.go [hunk 3]
```diff
@@ -69,40 +69,41 @@ func TestAPISearchRepo(t *testing.T) {
...
+			user:  {count: 21},
```
Comment (zeripath) [inline integrations/api_repo_test.go]:
```suggestion
			user:  {count: 19},
```

File: integrations/api_repo_test.go [hunk 4]
```diff
@@ -69,40 +69,41 @@ func TestAPISearchRepo(t *testing.T) {
...
+			user2: {count: 21}},
```
Comment (zeripath) [inline integrations/api_repo_test.go]:
```suggestion
			user2: {count: 19}},
```

**Comment by zeripath** [comment]
I was literally just doing this... 2 public repos were added between 1.9 and 1.8 so you need to change a few numbers in the test cases.

**Comment by zeripath** [comment]
Probably will need @mrsdizzie 's additional fix for collaborators on team repositories.

**Comment by mrsdizzie** [comment]
That was already back ported in 6968 and so things should just work when this change starts listing those repos in a users dashboard

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
Make lg-tm work

**Review by mrsdizzie — APPROVED** [review]

**Pull request merged**